Refactor to use Apache on Centos 7

gchaix opened the pull request.

gchaix committed:
Refactor to use Apache on Centos 7

gchaix committed:
Fix typos

jeffsheltren commented on README.md:
```diff
@@ -26,7 +27,7 @@ For full details, visit https://www.dokuwiki.org/dokuwiki
 ## Customizations
 
-The base container starts with the Alpine distribution, adding lighttpd and
+The base container starts with the Centos 7 distribution, adding Apahe and
```
Typo, should be: `Apache`

jeffsheltren commented on README.md:
```diff
@@ -40,14 +41,14 @@ You can comment out the line in Dockerfile that installs `php7-curl` and
-from `8080` to your desired port. Then, also edit the `EXPOSE` toward the end of
-Dockerfile, again changing `8080` to your new port.
+from `80` to your desired port. Then, also edit the `EXPOSE` toward the end of
+Dockerfile, again changing `808 to your new port.
```
`80` ?

jeffsheltren commented on app/Dockerfile:
```diff
@@ -0,0 +1,52 @@
+FROM centos:7
+
+ENV PATH=$PATH:/opt/rh/php71u/root/usr/bin
```
That /opt path shouldn't be needed since that's from SCL. Can we do PHP 7.2 or 7.3 instead?

gchaix committed:
Use PHP 7.2, move out of `app` to fix Quay

gchaix committed:
Use 'quiet' on wget so Quay build log is less noisy

gchaix committed:
Add security update during build

gchaix committed:
Initial helm template and yaml files

gchaix committed:
Tweaks to helm templates

gchaix committed:
Helm is making things too complicated right now

Here are the manually cleaned up yaml used to acutally run a deploy

gchaix commented:
Ignore the `./helm/*` for now.  It was being overly-complicated.  The correct yaml to use is in `./yaml`.

gchaix committed:
Apparently `yum update --security` doesn't get all security updates

jeffsheltren commented on Dockerfile:
```diff
@@ -1,30 +1,53 @@
-
+FROM centos:7
+
+ENV PATH=$PATH:/opt/rh/php72u/root/usr/bin
```
`/opt/rh/php72u/root/usr/bin` shouldn't be needed? IUS packages don't install into `/opt` AFAIK

jeffsheltren commented on Dockerfile:
```diff
@@ -1,30 +1,53 @@
+                  php72u-gmp php72u-pecl-apcu php72u-pecl-memcached \
+                  php72u-xml php72u-mbstring \
+                  php72u-gd php72u-zip php72u-mcrypt \
+                  wget curl git unzip yum-plugin-security" && \
```
Why `yum-plugin-security`? I don't that gets us much in a container.

jeffsheltren replied:
Oh, I see now the yum call below.. not sure how I feel about it, but not a blocker. Let's leave it while I ponder life, the universe, and gods of thunder.

gchaix committed:
Forgot the '-y'

gchaix committed:
Dump code in `/var/www/dokuwiki`

It will need to be copied into /var/www/html once the persistent volume
mounts.

gchaix committed:
Put persistent disk on `/var/www/html`

We'll need to copy the contents of `/var/www/dokuwiki` into the mount
after deployment.

gchaix committed:
Fix untar path

gchaix committed:
Remove unneded SCL-ism, fix docs

gchaix committed:
Fix build failure

gchaix committed:
Add Tag1 wildcard TLS and create an ingress controller

gchaix committed:
Bah, don't need that damn tempfile

gchaix committed:
Add temporary index file into the weboot

This allows the ready checks in k8s to succeed before we finish
installing doku.

gchaix committed:
Fix certificate label

Cert was added using:
`kubectl -n ingress-nginx create secret tls --cert tag1.crt --key tag1.key tag1-tls-wildcard`

gchaix committed:
Change URL, "docs" is already in use by gdocs

gchaix committed:
Make sure namespace exists

TODO: this currently whines on run about using "create", need to fix
